src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/utils.tsx

```diff
@@ -0,0 +1,480 @@
+import { prepareHistoryData } from '@Components/app/details/cdDetails/service'
+import { TemplateListDTO, TemplateListType } from '@Components/deploymentConfig/types'
+import {
+    AppEnvDeploymentConfigDTO,
+    AppEnvDeploymentConfigQueryParamsType,
+    AppEnvDeploymentConfigType,
+    ConfigMapSecretDataConfigDatumDTO,
+    ConfigResourceType,
+    DeploymentTemplateDTO,
+} from '@Pages/Applications/DevtronApps/service.types'
+import { DraftState } from '@Pages/Shared/ConfigMapSecret/ConfigMapSecret.types'
+
+import {
+    EnvironmentOptionType,
+    EnvResourceType,
+    AppEnvDeploymentConfigQueryParams,
+    DeploymentConfigParams,
+} from '../../AppConfig.types'
+
+/**
+ * Retrieves the draft data from the given configuration data object.
+ *
+ * @param configData - The configuration data object.
+ * @returns The draft data if available, otherwise the original data.
+ */
+export const getDraftData = (configData: ConfigMapSecretDataConfigDatumDTO): ConfigMapSecretDataConfigDatumDTO => {
+    if (configData?.draftMetadata) {
+        const parsedData = JSON.parse(configData.draftMetadata.data)
+        return { ...parsedData.configData[0], draftMetadata: configData.draftMetadata }
+    }
+
+    return configData
+}
+
+/**
+ * Combines two arrays of configuration data objects based on a 'name' key.
+ *
+ * @param primaryArray - The first array of configuration data objects.
+ * @param secondaryArray - The second array of configuration data objects.
+ * @returns The combined array of configuration data objects.
+ */
+export const mergeConfigDataArraysByName = (
+    primaryArray: ConfigMapSecretDataConfigDatumDTO[],
+    secondaryArray: ConfigMapSecretDataConfigDatumDTO[],
+): ConfigMapSecretDataConfigDatumDTO[][] => {
+    const dataMap = new Map<string, ConfigMapSecretDataConfigDatumDTO[]>()
+
+    const sortedPrimaryArray = primaryArray
+        .map(getDraftData)
+        .sort((a, b) => stringComparatorBySortOrder(a.name, b.name))
+
+    const sortedSecondaryArray = secondaryArray
+        .map(getDraftData)
+        .sort((a, b) => stringComparatorBySortOrder(a.name, b.name))
+
+    sortedPrimaryArray.forEach((item) => dataMap.set(item.name, [item, null]))
+    sortedSecondaryArray.forEach((item) => {
+        const key = item.name
+        if (dataMap.has(key)) {
+            dataMap.set(key, [dataMap.get(key)[0], item])
+        } else {
+            dataMap.set(key, [null, item])
+        }
+    })
+
+    return Array.from(dataMap.values())
+}
+
+const getObfuscatedData = (
+    codeEditorData: { [key: string]: string },
+    type: ConfigResourceType,
+    unAuthorized: boolean,
+) => {
+    const _codeEditorData = { ...codeEditorData }
+    if (type === ConfigResourceType.Secret && unAuthorized && _codeEditorData) {
+        Object.keys(_codeEditorData).reduce(
+            (acc, curr) => ({ ...acc, [acc[curr]]: Array(8).fill('*').join('') }),
+            _codeEditorData,
+        )
+    }
+    return _codeEditorData
+}
```

Comment:
does the BE gives the sensitive data on the FE?

Reply:
no BE gives empty string

Reply:
this is in disc with BE